### PR TITLE
Add support for using first marked used in compass order list to be used as primary

### PR DESCRIPTION
**Pull request by bugobliterator**
This PR allows one to disable a first compass and hence moving to use later compasses as primary if marked used.

### CODE REVIEW

**Comment by tridge** [comment]
need to test:
 - logging indexes
 - EK2/EK3 working with COMPASS_USE=0 and COMPASS_USE2=1
 - if possible, test DCM too with plane in SITL

File: libraries/AP_Compass/AP_Compass.cpp [hunk 1]
```diff
@@ -1706,6 +1706,16 @@ Compass::use_for_yaw(uint8_t i) const
     return _use_for_yaw[Priority(i)] && _learn.get() != LEARN_INFLIGHT;
 }
 
+uint8_t Compass::get_first_usable() const
+{
+    for (uint8_t i=0; i<COMPASS_MAX_INSTANCES; i++) {
+        if (use_for_yaw(i)) {
```
Comment (tridge) [inline libraries/AP_Compass/AP_Compass.cpp:1712]: need to directly use: _use_for_yaw[Priority(i)]

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by bugobliterator** [comment]
> need to test:
> 
> * logging indexes
verified correct
[logs.zip](https://github.com/ArduPilot/ardupilot/files/6517364/logs.zip)

> * mavlink raw_imu/scaled_imu2 indexes right
tested correct
> * EK2/EK3 working with COMPASS_USE=0 and COMPASS_USE2=1
yes both EKFs work, log in the zip
> * if possible, test DCM too with plane in SITL
also tested, log in the zip

File: libraries/AP_Compass/AP_Compass.h [hunk 2]
```diff
@@ -109,7 +109,7 @@ friend class AP_Compass_Backend;
     /// @returns heading in radians
     ///
     float calculate_heading(const Matrix3f &dcm_matrix) const {
-        return calculate_heading(dcm_matrix, 0);
+        return calculate_heading(dcm_matrix, get_first_usable());
```
Comment (tridge) [inline libraries/AP_Compass/AP_Compass.h:112]: should have a private variable uint8_t first_usable, which is updated once per update call

File: libraries/AP_Compass/AP_Compass.cpp [hunk 3]
```diff
@@ -1717,6 +1717,16 @@ Compass::use_for_yaw(uint8_t i) const
     return _use_for_yaw[Priority(i)] && _learn.get() != LEARN_INFLIGHT;
 }
 
+uint8_t Compass::get_first_usable() const
```
Comment (tridge) [inline libraries/AP_Compass/AP_Compass.cpp:1720]: add a comment please, explaining relationship to COMPASS_USE

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_Compass/AP_Compass.cpp (hunk 3 above)
Comment (bugobliterator) [reply libraries/AP_Compass/AP_Compass.cpp]: No longer applicable

File: libraries/AP_Compass/AP_Compass.h (hunk 2 above)
Comment (bugobliterator) [reply libraries/AP_Compass/AP_Compass.h]: done, please review. I have also retested by flying SITL copter.

**Review by tridge — APPROVED** [review]

**Pull request merged**